### PR TITLE
Fix release generation

### PR DESCRIPTION
**Pull request by fernandofcampos**
## Purpose of Changes and their Description

The purpose of this change is to fix the release generation and only create a new release when a tag is added.

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

https://linear.app/alloralabs/issue/PROTO-2135/fix-release-generation

## Are these changes tested and documented?

- [x] If tested, please describe how. If not, why tests are not needed.

Tested locally using https://github.com/nektos/act

- [ ] If documented, please describe where. If not, describe why docs are not needed.
- [ ] Added to `Unreleased` section of `CHANGELOG.md`?

## Still Left Todo

*Fill this out if this is a Draft PR so others can help.*

### CODE REVIEW

File: .github/workflows/goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: goreleaser
+
+on:
+  push:
+    tags:
+      - '*'
```
Comment (okedeji) [inline .github/workflows/goreleaser.yml:6]: for sanity, maybe we want to ensure it only releases for tags that start with `v*`.

File: .github/workflows/goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: goreleaser
+
+on:
+  push:
+    tags:
+      - 'v*'
```
Comment (spooktheducks) [inline .github/workflows/goreleaser.yml:6]: nitpick, but if proper regex is possible in this field, `v\d+\.\d+\.\d+` would be better

File: .github/workflows/goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      -
+        name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22.2
```
Comment (spooktheducks) [inline .github/workflows/goreleaser.yml:24]: Not addressable here but we should consider moving to 1.23 soon if possible

File: .goreleaser.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      - arm64
+
+    # Environment variables
+    env:
+      - CGO_ENABLED=0
+
+archives:
+  - format: tar.gz
+    name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
+    files:
```
Comment (spooktheducks) [inline .goreleaser.yaml:35]: I'm guessing it automatically includes the binary as well right?

**Review by spooktheducks — APPROVED** [review]

File: .goreleaser.yaml (hunk 4 above)
Comment (fernandofcampos) [reply .goreleaser.yaml:35]: It does

**Comment by relyt29** [comment]
PLEASE REBASE THIS BRANCH ONTO DEV AGAIN, AS DEV HAS BEEN PUSHED TO A NEW HEAD THAT REMOVES A BAD COMMIT

```bash 
git fetch 
git rebase -i origin/dev
```

@fernandofcampos

**Review by kpeluso — APPROVED** [review]

**Review by okedeji — APPROVED** [review]
lgtm

**Pull request merged**